ebu_tt_live/documents/ebutt3.py

```diff
@@ -717,12 +717,20 @@ def extract_segment(self, begin=None, end=None, sequence_number=None, discard=Fa
         for doc in affected_documents:
             doc_ending = doc.resolved_end_time
             if end is not None:
-                if end < doc_ending:
+                if doc_ending is None or end < doc_ending:
                     doc_ending = end
             # Check only til resolved end, otherwise there will be unwanted parallel elements
-            doc_segment = doc.extract_segment(begin=begin, end=doc_ending, deconflict_ids=True)
-
-            document_segments.append(doc_segment)
+            try:
+                doc_segment = doc.extract_segment(begin=begin, end=doc_ending, deconflict_ids=True)
+                document_segments.append(doc_segment)
+            except Exception as err:
+                log.error(
+                    'Error extracting document segment from {}__{}'.format(
```

Comment:
well you were unhappy about stacktraces on the console so I changed log.exception to log.error, which would not give you a stacktrace but I did log the exception instance in addition to at least give a clue where to start.